pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -1084,7 +1084,10 @@ func (m *kubeGenericRuntimeManager) computeInitContainerActions(pod *v1.Pod, pod
 
```

Comment:
computeInitContainerActions looks like a new code path only enabled when the SidecarContainers feature is enabled?

taking a step back, should we only take this new code path in pods which actually have made use of the sidecar feature? ideally, bugs in this new code path would only impact pods that have specifically opted into using the new sidecar feature, and all other pods would have their behavior stay the same.

Reply:
I would like to remove the old code path while updating the new one once the SidecarContainers feature reaches GA, since there is a lot of shared logic for the init container's actions.

Reply:
> I would like to remove the old code path while updating the new one once the SidecarContainers feature reaches GA, since there is a lot of shared logic for the init container's actions.

The number one priority is to ensure that pods that didn't opt into the new feature are not impacted by bugs in the new code path. This is the second time the sidecar feature has regressed, so I think it's important to leave the old code path active for pods which have not opted in until the new code path has proven it works correctly over multiple releases.

Reply:
I agree that it should have been implemented in that way, but it might be a bit late now since the sidecar wg is considering to promote this feature to GA in the next release...

cc @SergeyKanzhelev

Reply:
> it might be a bit late now since the sidecar wg is considering to promote this feature to GA in the next release...

I do not agree. The presence of a regression in this code path should make us hesitant to make this code path unconditional. Recovering compatibility for 1.29-1.31 should be our top priority. Reviewing these changes and closing test gaps should be our second priority. Promotion to GA should come after all of those.

Reply:
ok, I'll create a PR for that to ensure that the minimize the impact of the new code path.